boards/nxp/s32z2xxdc2/doc/index.rst

```diff
@@ -64,6 +64,10 @@ The boards support the following hardware features:
 +-----------+------------+-------------------------------------+
 | QSPI      | on-chip    | flash                               |
 +-----------+------------+-------------------------------------+
+| STM       | on-chip    | counter                             |
++-----------+------------+-------------------------------------+
+| PIT       | on-chip    | counter                             |
++-----------+------------+-------------------------------------+
```

Comment:
You guys may consider becoming early adopters of the `.. zephyr:board-supported-hw::` that would basically keep the table always up-to-date for you :)

You would basically just need to replace the table with the aforementioned directive .

![image](https://github.com/user-attachments/assets/5d0e7af0-fc7b-4f5f-b999-70646b8c4691)

Reply:
I saw your pr before and it looks great! nice addition. If it's okay, I will open a pr to adopt this for all our boards in one go, because I noticed the other boards also might be missing some entries on the hw support table